### PR TITLE
Hide reason text in `status-subscription`

### PR DESCRIPTION
**Pull request by fregante**
Extracted from a discussion in https://github.com/refined-github/refined-github/pull/6667 cc @FloEdelmann 

This text doesn't add anything more to the state shown by the feature, except when "all" is selected, because it can be one of:

- You’re receiving notifications because you’re watching this repository.
- You’re receiving notifications because you authored the thread.
- You’re receiving notifications because you commented.
- You’re receiving notifications because you were mentioned.
- _who knows what else_

## URL

Here

### CODE REVIEW

**Comment by fregante** [comment]
Also verified:

Screenshot 2

File: source/features/status-subscription.tsx [hunk 1]
```diff
@@ -88,6 +91,10 @@ function addButton(subscriptionButton: HTMLButtonElement): void {
 	// Remove it only if the form was successfully added
 	originalId.remove();
 	subscriptionButton.hidden = true;
+
+	if (status !== 'all') {
```
Comment (FloEdelmann) [inline source/features/status-subscription.tsx:95]:
```suggestion
	// Different reasons lead to the "all notifications enabled" state #6684
	if (status !== 'all') {
```

**Review by FloEdelmann — APPROVED** [review]
Makes sense to me :+1:

File: source/features/status-subscription.tsx [hunk 2]
```diff
@@ -98,7 +98,7 @@ function addButton(subscriptionButton: HTMLButtonElement): void {
 }
 
 function init(signal: AbortSignal): void | false {
-	observe('button[data-thread-subscribe-button]', addButton, {signal});
+	observe('button[data-thread-subscribe-button]:not([disabled])', addButton, {signal});
```
Comment (FloEdelmann) [inline source/features/status-subscription.tsx:101]: Maybe mention this in the "Test URLs" section?

File: source/features/status-subscription.tsx (hunk 1 above)
Comment (fregante) [reply source/features/status-subscription.tsx:95]: How about

```suggestion
	// 'all' can have many reasons, but the other two don't add further information #6684
	if (status !== 'all') {
```

**Review by FloEdelmann — APPROVED** [review]
Again :wink:

**Pull request merged**